main.cpp

```diff
@@ -7,22 +7,36 @@
 int main(int argc, char *argv[]) {
     /* Start the GUI work */
     auto app = Gtk::Application::create(argc, argv, "br.com.jeanhertel.adriconf");
-    try {
-        DRIQuery check;
+
+    char *protocol = std::getenv("XDG_SESSION_TYPE");
+
+    DRIQuery check;
+    if (!check.isSystemSupported()) return 1;
```

Comment:
Why no message is shown to the user when we exit?
Also try to always add braces around statements, so it becomes more clear what code belongs to it.

Reply:
we are giving a message in `isSystemSupported` function in DRIQuery.cpp and HelpersWayland.cpp